src/Symfony/Component/HttpKernel/Event/ExceptionEvent.php

```diff
@@ -24,6 +27,64 @@
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class ExceptionEvent extends GetResponseForExceptionEvent
+class ExceptionEvent extends RequestEvent
 {
+    /**
+     * The exception object.
+     *
+     * @var \Exception
+     */
+    private $exception;
+
+    /**
+     * @var bool
+     */
+    private $allowCustomResponseCode = false;
+
+    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, \Exception $e)
+    {
+        parent::__construct($kernel, $request, $requestType);
+
+        $this->setException($e);
+    }
+
+    /**
+     * Returns the thrown exception.
+     *
+     * @return \Exception The thrown exception
+     */
+    public function getException()
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4